source/simulator/lateral_averaging.cc

```diff
@@ -466,7 +466,24 @@ namespace aspect
         std::vector<double> field_values;
     };
   }
+  template <int dim>
```

Comment:
please add three empty lines before the template. this is our usual style to make sure the file remains readable.